mobile/native/src/api.rs

```diff
@@ -246,16 +248,46 @@ pub fn subscribe(stream: StreamSink<event::api::Event>) {
     event::subscribe(FlutterSubscriber::new(stream))
 }
 
-/// Wrapper for Flutter purposes - can throw an exception.
-pub fn run_in_flutter(seed_dir: String, fcm_token: String) -> Result<()> {
-    let result = if !IS_INITIALISED.try_get().unwrap_or(&false) {
-        run(seed_dir, fcm_token, IncludeBacktraceOnPanic::Yes)
-            .context("Failed to start the backend")
-    } else {
-        Ok(())
-    };
-    IS_INITIALISED.set(true);
-    result
+pub fn run(seed_dir: String, fcm_token: String) -> Result<()> {
+    match IS_INITIALISED.try_get() {
+        None => {
+            let (tx_websocket, _rx) = broadcast::channel(10);
+            tracing::info!("Staring backend");
+            run_internal(
+                seed_dir,
+                fcm_token,
+                tx_websocket.clone(),
+                IncludeBacktraceOnPanic::Yes,
+            )
+            .context("Failed to start the backend")?;
+
+            IS_INITIALISED.set(tx_websocket);
+
+            Ok(())
+        }
+        Some(tx_websocket) => {
+            // In case of a hot-restart we do not start the node again as it is already running.
+            // However, we need to re-send the authentication message to get the initial data from
+            // the coordinator and trigger a new user login event.
+            tracing::info!("Re-sending authentication message");
+            let signature = ln_dlc::get_node_key()
+                .sign_ecdsa(create_sign_message(commons::AUTH_SIGN_MESSAGE.to_vec()));
+            let signature = commons::Signature {
+                pubkey: ln_dlc::get_node_pubkey(),
+                signature,
+            };
+
+            let runtime = crate::state::get_or_create_tokio_runtime()?;
+            runtime.block_on(async {
+                tx_websocket.send(OrderbookRequest::Authenticate {
+                    fcm_token: Some(fcm_token),
+                    signature,
+                })
+            })?;
+
+            Ok(())
+        }
```

Comment:
🤔 I don't like having to duplicate this logic just so that we can do a hot restart.

I would try to:

- Remove the `run`-`run_internal` separation and just have a single `run` function.
- Make it so that we can drop the orderbook WS task by returning a handle to it and keeping it in a static variable.
- On restart, always call `run` and, if the node is already initialised, clean up the old task and reconnect to the coordinator.

Or maybe we can refactor it a different way. But duplicating this logic is not very nice.

Reply:
I improved it a little bit, but I guess it can still be done nicer. Leaving it as is for now, as this is not super critical.